### PR TITLE
Jacoco was added. Maven parent project now has setted utf-8 encoding.…

### PR DESCRIPTION
**Pull request by sledzgrzegorz**
JACOCO was added. UTF-8 in parent maven project was fixed.
How to use:
run: mvn clean package
go to: {$moduleName}/target/site/jacoco-ut
open index.html in your web browser

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,17 +1,80 @@
...
+        <module>battleship-logic</module>
+        <module>battleship-service</module>
+    </modules>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.5.1</version>
```
Comment (LIttleAncientForestKami) [inline pom.xml]: All numbers should be in properties.

File: pom.xml [hunk 2]
```diff
@@ -1,17 +1,80 @@
...
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
```
Comment (LIttleAncientForestKami) [inline pom.xml]: Do you plan on EVER having source and target apart? If not, use one property for both.

File: pom.xml [hunk 3]
```diff
@@ -1,17 +1,80 @@
...
+                    </execution>
+                    <!--
+                        Ensures that the code coverage report for unit tests is created after
+                        unit tests have been run.
+                    -->
+                    <execution>
+                        <id>post-unit-test</id>
+                        <phase>test</phase>
+                        <goals>
+                            <goal>report</goal>
```
Comment (LIttleAncientForestKami) [inline pom.xml]: Here! Here, exactly!

As you noticed, it's a plugin tied to reporting phase. I'd like it's report to be tied to `mvn site` run, and available from your project site.

File: pom.xml [hunk 4]
```diff
@@ -1,17 +1,80 @@
...
+                            <outputDirectory>${project.reporting.outputDirectory}/jacoco-ut</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (LIttleAncientForestKami) [inline pom.xml]: :+1:

**Comment by LIttleAncientForestKami** [comment]
Almost good to go. Please address properties instead of numbers and add how to run info to documentation (Wiki / README) and bind JaCoCo report with `mvn site` for your project.

File: pom.xml (hunk 3 above)
Comment (LukaszCzubernat) [inline pom.xml]: Did you think about add some configuration to jacoco plugin which will chceck your code during build proccess?
Maybe add some rules which your code should follow?
Like:
<execution>
                        <id>default-check</id>
                        <goals>
                            <goal>check</goal>
                        </goals>
                        <configuration>
                            <rules>
                                <rule>
                                    <element>BUNDLE</element>
                                    <excludes>
                                    </excludes>
                                    <limits>
                                        <limit>
                                            <counter>INSTRUCTION</counter>
                                            <value>COVEREDRATIO</value>
                                            <minimum>0.90</minimum>
                                        </limit>
                                        <limit>
                                            <counter>LINE</counter>
                                            <value>COVEREDRATIO</value>
                                            <minimum>0.60</minimum>
                                        </limit>
                                    </limits>
                                </rule>
                            </rules>
                        </configuration>
                    </execution>

**Comment by LIttleAncientForestKami** [comment]
@egolesor @sledzgrzegorz were mine and @LukaszCzubernat's remarks incorporated? Ignored?

**Pull request merged**